src/App/Order/Calculator/UPS/UPSDeliveryTypeCalculator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\App\Order\Calculator\UPS;
+
+use MyParcelNL\Pdk\App\Order\Calculator\AbstractPdkOrderOptionCalculator;
+use MyParcelNL\Pdk\Base\Service\CountryCodes;
+use MyParcelNL\Pdk\Shipment\Model\DeliveryOptions;
+
+class UPSDeliveryTypeCalculator extends AbstractPdkOrderOptionCalculator
+{
+    /**
+     * @inheritDoc
+     */
+    public function calculate(): void
+    {
+        $deliveryOptions = $this->order->deliveryOptions;
+        $cc              = $this->order->shippingAddress->cc;
+
+        switch ($deliveryOptions->deliveryType) {
```

Comment:
Deze calculator is denk ik met het oog op de toekomst gebouwd, maar een switch met maar 1 case en waar maximaal ook maar 2 cases kunnen komen, met in de case bovendien een if (die iets gebruikt dat altijd wordt opgehaald, regel 19, ook als het niet gebruikt wordt, wat vermoedelijk meestal is), daarom wil ik hem op zich niet afkeuren, maar deze opmerking is om te benadrukken dat het goed is om de code simpel te houden (een enkele if hier bijvoorbeeld) zolang je de tierelantijnen niet nodig hebt. Want als de volgende developer sneller begrijpt wat er gebeurt, kan die de tierelantijnen zelf toevoegen, indien nodig, later.

Reply:
Ik heb hem ingekort. De switch was inderdaad bedoelt om rekening te houden met de toekomst.